library/src/main/java/com/pokegoapi/api/PokemonGo.java

```diff
@@ -72,12 +73,14 @@
 	 * @param credentialProvider the credential provider
 	 * @param client             the http client
 	 * @param time               a time implementation
+	 * @param needSleep			 need sleep after creating settings
 	 * @throws LoginFailedException  When login fails
 	 * @throws RemoteServerException When server fails
+	 * @throws InterruptedException When Thread.sleep fails
```

Comment:
`Thread.sleep` doesn't fail. `InterruptedException` happens when another thread interrupts the current one. Although Java likes to encode unexceptional situations in exceptions, this is not a failure mode.